org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/lookup/Scope.java

```diff
@@ -3027,6 +3027,13 @@ public final ReferenceBinding getJavaUtilIterator() {
 	*/
 	public final ReferenceBinding getMemberType(char[] typeName, ReferenceBinding enclosingType) {
 		ReferenceBinding memberType = findMemberType(typeName, enclosingType);
```

Comment:
I wonder if the change can be made inside `findMemberType`, near the location where currently `null` is returned?

Otherwise, I suggest to let the new code below check if `enclosingType.isHierarchyBeingConnected()` to ensure that the additional code is invoked only in this particular situation.

Reply:
> I wonder if the change can be made inside findMemberType, near the location where currently null is returned?

I tried that, but some compiler regression tests fail.

> Otherwise, I suggest to let the new code below check if enclosingType.isHierarchyBeingConnected() to ensure that the additional code is invoked only in this particular situation.

It has been fixed.